client/src/partials/cash/modals/invoices.modal.js

```diff
@@ -27,26 +26,34 @@ function CashInvoiceModalController(Debtors, debtorId, invoiceIds, ModalInstance
   vm.cancel = ModalInstance.dismiss;
   vm.submit = submit;
 
-  // ui-grid options
   vm.gridOptions = {
     appScopeProvider : vm,
-    enableSorting : false,
-    multiSelect : true,
-    showGridFooter : true,
-    onRegisterApi : bindGridApi,
+    multiSelect: true,
+    fastWatch: true,
+    flatEntityAccess: true,
+    onRegisterApi : onRegisterApi,
+    enableColumnMenus: false,
     columnDefs : [
-      { name : 'reference', enableHiding : false },
-      { name : 'balance', cellFilter: 'currency:' + Session.enterprise.currencyId, enableHiding : false, },
-      { name : 'date', cellFilter: 'date', enableHiding: false }
+      { name : 'reference'},
+      { name : 'balance', cellFilter: 'currency:' + Session.enterprise.currencyId},
+      { name : 'date', cellFilter: 'date' }
     ],
     minRowsToShow : 10
   };
 
+  function selectionChangeCallback() {
+    vm.rows = vm.getSelectedRows();
+  }
+
   // in order to use controllerAs syntax, we need to import the entire grid API
   // into the controller scope to bind the getSelectedRows method.
-  function bindGridApi(api) {
+  function onRegisterApi(api) {
```

Comment:
This modal was written about four months before the `GridSelection` service was used.  I tried to keep my clientside changes minimal in this PR to focus on the server posting logic.  Most of the changes on the client were attempted bug fixes or removing old code.  In a separate PR, I'll come through and finish the client refactor.